client/my-sites/upgrades/cart/cart-item.jsx

```diff
@@ -85,6 +95,41 @@ module.exports = React.createClass( {
 		return info;
 	},
 
+	handleDomainVolumeSelection: function( event ) {
+		event.preventDefault();
+		let volume = parseInt( event.target.value );
+		upgradesActions.setVolume( this.props.cartItem, volume );
+
+		this.recordEvent( 'changeVolume', volume );
+	},
+
+	getVolumeOptions: function() {
+		return range( 1, 6 ).map( number =>
+			<option key={ number } value={ number }>
+				{ this.translate( '%(number)s year', '%(number)s years', { args: { number }, count: number } ) }
+			</option>
+		);
+	},
+
+	domainVolumeSelection: function() {
+		if ( ! config.isEnabled( 'upgrades/cart/multi-year-domain' ) ) {
+			return null;
+		}
+
+		if ( this.isBundlePlanApplied() || this.props.cartItem.free_trial || ! isDomainRegistration( this.props.cartItem ) ) {
+			return null;
+		}
+
+		return (
+			<select name="product-domain-volume"
+							className="product-domain-volume"
+							onChange={ this.handleDomainVolumeSelection }
+							value={ this.props.cartItem.volume }>
```

Comment:
:+1: I use 2 space tabs to use space efficiently, Webstorm wanted to align for that. :+1: